boa/src/syntax/parser/mod.rs

```diff
@@ -43,12 +51,16 @@ impl fmt::Display for ParseError {
                 write!(f, "Expected expression '{}', got '{}'", expected, actual)
             }
             ParseError::UnexpectedKeyword(keyword) => write!(f, "Unexpected keyword: {}", keyword),
+            ParseError::Unexpected(tok, msg) => write!(f, "Unexpected Token {} {}", tok, msg),
             ParseError::AbruptEnd => write!(f, "Abrupt End"),
+            ParseError::General(msg) => write!(f, "{}", msg),
+            ParseError::NormalEOF => write!(f, "EOF"),
+            ParseError::RangeError => write!(f, "RangeError!"),
```

Comment:
I think we should also add line/column information for at least `UnexpectedKeyword` and `Unexpected`, and maybe also in `AbruptEnd` and `RangeError`, since they don't give very good debugging information.

Reply:
What do you guys think about this? The `AbruptEnd` variant will only be triggered when the parser has no extra tokens to retrieve (EOF, until we mess things up with streams xD), so no need to add line information here. `General` is only used once, and I added position information, and I already added position information to `UnexpectedKeyword`. Should we add it to `RangeError`?

I think we should improve the error messages of `RangeError` and `AbruptEnd`, though, and we might want to revisit the need of `General`, that it's only used once.